### PR TITLE
Create a bill batching function with machine updates

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Refactor create batch functions to run on the server side.](https://trello.com/c/AEX5OyEk/1120-refactor-create-batch-functions-to-run-on-the-server-side)

File: lib/cashbox-batches.js [hunk 1]
```diff
@@ -18,6 +19,32 @@ function createCashboxBatch (deviceId, cashboxCount) {
   })
 }
 
+function updateMachineWithBatch (machineContext, oldCashboxCount) {
+  if (!_.has(['deviceId', 'cashbox', 'cassettes'], machineContext) && !_.inRange(constants.CASH_OUT_MINIMUM_AMOUNT_OF_CASSETTES, constants.CASH_OUT_MAXIMUM_AMOUNT_OF_CASSETTES + 1, _.size(machineContext.cassettes)))
```
Comment (josepfo) [inline lib/cashbox-batches.js]: This line seems too long

File: lib/cashbox-batches.js [hunk 2]
```diff
@@ -18,6 +19,32 @@ function createCashboxBatch (deviceId, cashboxCount) {
...
+    const q2 = t.none(`UPDATE bills SET cashbox_batch_id=$1 FROM cash_in_txs
+      WHERE bills.cash_in_txs_id = cash_in_txs.id AND
+      cash_in_txs.device_id = $2 AND 
+      bills.cashbox_batch_id IS NULL`, [batchId, deviceId])
+    const q3 = t.none(`UPDATE devices SET cashbox=$1, cassette1=$2, cassette2=$3, cassette3=$4, cassette4=$5 WHERE device_id=$6`, [
+      machineContext.cashbox,
+      machineContext.cassettes[0],
+      machineContext.cassettes[1],
+      machineContext.cassettes[2],
+      machineContext.cassettes[3],
```
Comment (josepfo) [inline lib/cashbox-batches.js:42]: Do we have 0 as a default value for each cassette? For devices that only have 2 for instance

**Review by josepfo — CHANGES_REQUESTED** [review]

**Comment by josepfo** [comment]
Has conflicts

File: new-lamassu-admin/src/pages/Maintenance/Wizard/Wizard.js [hunk 3]
```diff
@@ -49,6 +69,8 @@ const Wizard = ({ machine, cashoutSettings, locale, onClose, save, error }) => {
     })
   }
 
+  console.log('AAA', CASSETTE_FIELDS)
```
Comment (josepfo) [inline new-lamassu-admin/src/pages/Maintenance/Wizard/Wizard.js]: You missed this one

File: new-lamassu-admin/src/pages/Maintenance/Wizard/Wizard.js [hunk 4]
```diff
@@ -11,6 +12,14 @@ const MODAL_WIDTH = 554
 const MODAL_HEIGHT = 520
 const CASHBOX_DEFAULT_CAPACITY = 500
 
+// Maximum number of cassettes, to create the necessary fields for the back-end
+const MAX_NUMBER_OF_CASSETTES = 4
```
Comment (josepfo) [inline new-lamassu-admin/src/pages/Maintenance/Wizard/Wizard.js]: Wondering if this should live on l-s/new-lamassu-admin/src/utils/constants.js

File: new-lamassu-admin/src/pages/Maintenance/CashCassettes.js [hunk 5]
```diff
@@ -299,7 +272,7 @@ const CashCassettes = () => {
       <TitleSection
         title="Cash Cassettes"
         button={{
-          text: 'Cashbox history',
+          text: 'Cashbox history a',
```
Comment (josepfo) [inline new-lamassu-admin/src/pages/Maintenance/CashCassettes.js]: This is probably an unintentional change

**Review by josepfo — CHANGES_REQUESTED** [review]

**Review by josepfo — APPROVED** [review]

**Pull request merged**